### PR TITLE
Features/proxy generatoaddress

### PR DESCRIPTION
**Pull request by filecommander**
Added a generatetoaddress in proxy.  This lets the proxy/tests/tests.sh run without user having to mine 1 block

### CODE REVIEW

**Comment by Kexkey** [comment]
I'd like the arguments to all be optional: if no `nbblocks` provided, use 1 and if no `address` supplied, get a new one from the spending wallet.

Thanks!!

File: doc/openapi/v0/cyphernode-api.yaml [hunk 1]
```diff
@@ -1822,6 +1822,55 @@ paths:
...
+      operationId: "generateToAddress"
+      requestBody:
+        description: "Number of blocks, destination address, max number of iterations to try"
+        required: true
+        content:
+          application/json:
+            schema:
+              type: "object"
+              properties:
+                nbBlocks:
```
Comment (Kexkey) [inline doc/openapi/v0/cyphernode-api.yaml:1904]: Make the arguments optional and the request HTTP-GET compatible

**Review by Kexkey — CHANGES_REQUESTED** [review]

File: doc/openapi/v0/cyphernode-api.yaml (hunk 1 above)
Comment (filecommander) [reply doc/openapi/v0/cyphernode-api.yaml:1904]: Done!

File: proxy_docker/app/script/blockchainrpc.sh [hunk 2]
```diff
@@ -114,3 +115,26 @@ bitcoin_estimatesmartfee() {
...
   return $?
 }
+
+
+bitcoin_generatetoaddress() {
+  trace "Entering bitcoin_generatetoaddress()..."
+
+  local nbblocks=$(echo ${1} | jq '.nbblocks // 1') # Optional - Default 1
+  local address=$(echo ${1} | jq '.address' // empty) # Optional - getnewadress from spender wallet
+  local maxtries=$(echo ${1} | jq '.maxtries // 1000000')  # Optional - use Core default
```
Comment (filecommander) [inline proxy_docker/app/script/blockchainrpc.sh:124]: I hard coded core default value - not using core's actual or futur default value.   Would it be better to leave it empty if not specified ?

File: cyphernodeconf_docker/templates/gatekeeper/api.properties [hunk 3]
```diff
@@ -72,6 +72,7 @@ action_removefrombatch=spender
 action_listbatchers=spender
 action_getbatcher=spender
 action_getbatchdetails=spender
+action_generatetoaddress=spender
```
Comment (Kexkey) [inline cyphernodeconf_docker/templates/gatekeeper/api.properties]: It's actually `bitcoin_generatetoaddress`
This doesn't work when request goes through the gatekeeper.
Maybe we should add going through the gatekeeper in all the tests

[the remark this answers is not in the document]
Comment (Kexkey) [reply cyphernodeconf_docker/templates/gatekeeper/api.properties]: Oh I know, we should add the new endpoint to `api_auth_docker/tests/test-gatekeeper.sh`

**Review by Kexkey — CHANGES_REQUESTED** [review]

File: cyphernodeconf_docker/templates/gatekeeper/api.properties (hunk 3 above)
Comment (filecommander) [reply cyphernodeconf_docker/templates/gatekeeper/api.properties]: OK, fixed it

**Comment by filecommander** [comment]
All changes made

**Pull request merged**